Cleanup datasets and demos

jacobmontiel opened the pull request:
Changes proposed in this pull request:

This PR is intended to clean up the project in two main aspects:

1. Remove datasets from the project as now they are placed in a dedicated repository: https://github.com/scikit-multiflow/streaming-datasets
    1. All references to dataset file were replaced with generators whenever possible or with the corresponding URL for the dataset in the repository
    2. The file used in functional tests for `FileStream` and `DataStream` was replaced with a smaller version exclusive for testing.
2. Make the `demos` module private (`_demos`) as now most of the methods include usage examples in their docstrings. This is the first step in the eventual removal of this submodule.

---

### Checklist
#### Implementation
- [x] Implementation is correct (it performs its intended function).
- [x] Code complies with PEP-8 and is consistent with the framework.
- [x] Code is properly documented.
- [x] PR description covers ALL the changes performed.
- [x] Files changed (update, add, delete) are in the PR's scope (no extra files are included).

#### Tests
- [ ] New functionality is tested.
- [x] Tests are created for the new functionality or existing tests are updated accordingly.
- [x] ALL tests pass with no errors.
- [x] CI/CD pipelines run with no errors.
- [x] Test Coverage is maintained (coverage may drop by no more than 0.2%).

smastelini commented on src/skmultiflow/_demos/_test_mol.py:
```diff
@@ -23,7 +23,8 @@ def demo():
 
     # Setup the file stream
-    stream = FileStream("../data/datasets/music.csv", 0, 6)
+    stream = FileStream("https://raw.githubusercontent.com/scikit-multiflow/streaming-datasets/"
+                        "master/", 0, 6)
```
```suggestion
    stream = FileStream("https://raw.githubusercontent.com/scikit-multiflow/streaming-datasets/"
                        "master/music.csv", 0, 6)
```

jacobmontiel replied:
Fixed in e8b8f3fd3bb60bd1e6ad657e3fe45da11ac0a8d2

smastelini commented on src/skmultiflow/_demos/_test_stream_speed.py:
```diff
@@ -9,7 +9,8 @@ def demo():
     """
     # Setup the stream
-    # stream = FileStream("../data/datasets/covtype.csv", -1, 1)
+    # stream = FileStream("https://raw.githubusercontent.com/scikit-multiflow/streaming-datasets/"
+    #                     "master/covtype.csv")
```
Are the previously passed `-1, 1` the  default values of `FileStream`?

smastelini replied:
I just checked and the answer is yes :)

I'm marking that one as solved.

smastelini commented on src/skmultiflow/data/file_stream.py:
```diff
@@ -39,7 +39,8 @@ class FileStream(Stream):
     >>> from skmultiflow.data.file_stream import FileStream
     >>> # Setup the stream
-    >>> stream = FileStream('skmultiflow/data/datasets/sea_stream.csv')
+    >>> stream = FileStream("https://raw.githubusercontent.com/scikit-multiflow/"
+    ...                     "master/sea_stream.csv")
```
I believe the link is incomplete (the repository containing the datasets is missing).

jacobmontiel replied:
Fixed in e8b8f3fd3bb60bd1e6ad657e3fe45da11ac0a8d2

smastelini commented on src/skmultiflow/lazy/sam_knn.py:
```diff
@@ -67,13 +67,15 @@ class SAMKNNClassifier(BaseSKMObject, ClassifierMixin):
     >>> from skmultiflow.evaluation import EvaluatePrequential
     >>> # Setup the File Stream
-    >>> stream = FileStream("moving_squares.csv")
+    >>> stream = FileStream("https://raw.githubusercontent.com/scikit-multiflow/"
+    ...                     "master/moving_squares.csv")
```
Same situation here.

jacobmontiel replied:
Fixed in e8b8f3fd3bb60bd1e6ad657e3fe45da11ac0a8d2

smastelini commented on src/skmultiflow/meta/multi_output_learner.py:
```diff
@@ -40,7 +40,8 @@ class MultiOutputLearner(BaseSKMObject, ClassifierMixin, MetaEstimatorMixin, Mul
     >>> from sklearn.linear_model import Perceptron
     >>> # Setup the file stream
-    >>> stream = FileStream("skmultiflow/data/datasets/music.csv", 0, 6)
+    >>> stream = FileStream("https://raw.githubusercontent.com/scikit-multiflow/"
+    ...                     "master/moving_squares.csv", 0, 6)
```
The repository does not appear in the link.

jacobmontiel replied:
Fixed in e8b8f3fd3bb60bd1e6ad657e3fe45da11ac0a8d2

smastelini commented:
Thanks for the changes @jacobmontiel. The package size is going to be reduced considerably!

I let some minor comments in some parts of the changed code.

smastelini commented:
I did a fast search locally and found some references to the (now) outdated `datasets` folder in:

* `.gitattributes`
* `docs/user-guide/quick-start.rst` (is that one generated with the documentation?)
* `MANIFEST.in`

Should these references be removed from the mentioned files in this PR?

jacobmontiel commented:
> I did a fast search locally and found some references to the (now) outdated `datasets` folder in:
> 
> * `.gitattributes`
> * `docs/user-guide/quick-start.rst` (is that one generated with the documentation?)
> * `MANIFEST.in`
> 
> Should these references be removed from the mentioned files in this PR?

I removed the references from `MANIFEST.in` and updated the example in `quick-start.rst` in e8b8f3fd3bb60bd1e6ad657e3fe45da11ac0a8d2.

I am keeping the reference in the `.gitattributes` since it applies to the repository history that will still have those files for backward compatibility.

smastelini commented:
Thanks for the last changes @jacobmontiel!

> I am keeping the reference in the .gitattributes since it applies to the repository history that will still have those files for backward compatibility.

Nice catch! I believe the PR is ready to be merged. Nonetheless, the CI/CD tests are failing, though the problem does not seem to be related to this PR. Do you have any leads about the source of the problem?
